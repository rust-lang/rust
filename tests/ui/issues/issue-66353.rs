// #66353: ICE when trying to recover from incorrect associated type

trait _Func<T> {
    fn func(_: Self);
}

trait _A {
    type AssocT;
}

fn main() {
    _Func::< <() as _A>::AssocT >::func(());
    //~^ ERROR the trait bound `(): _A` is not satisfied
    //~| ERROR the trait bound `(): _Func<_>` is not satisfied
}
