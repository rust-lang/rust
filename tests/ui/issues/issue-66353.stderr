error[E0277]: the trait bound `(): _A` is not satisfied
  --> $DIR/issue-66353.rs:12:15
   |
LL |     _Func::< <() as _A>::AssocT >::func(());
   |               ^^ the trait `_A` is not implemented for `()`
   |
help: this trait has no implementations, consider adding one
  --> $DIR/issue-66353.rs:7:1
   |
LL | trait _A {
   | ^^^^^^^^

error[E0277]: the trait bound `(): _Func<_>` is not satisfied
  --> $DIR/issue-66353.rs:12:41
   |
LL |     _Func::< <() as _A>::AssocT >::func(());
   |     ----------------------------------- ^^ the trait `_Func<_>` is not implemented for `()`
   |     |
   |     required by a bound introduced by this call
   |
help: this trait has no implementations, consider adding one
  --> $DIR/issue-66353.rs:3:1
   |
LL | trait _Func<T> {
   | ^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0277`.
